src/main/java/com/internet/shop/controllers/cart/RemoveProductFromCartController.java

```diff
@@ -0,0 +1,29 @@
+package com.internet.shop.controllers.cart;
+
+import com.internet.shop.lib.Injector;
+import com.internet.shop.service.ProductService;
+import com.internet.shop.service.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class RemoveProductFromCartController extends HttpServlet {
+    private static final Long USER_ID = 1L;
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final ProductService productService = (ProductService) injector
+            .getInstance(ProductService.class);
+    private final ShoppingCartService shoppingCartService =
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String productId = req.getParameter("id");
+        Long id = Long.valueOf(productId);
+        shoppingCartService.deleteProduct(shoppingCartService.getByUserId(USER_ID),
+                productService.get(id));
```

Comment:
This will remove a product from the cart by product id, not by the index of the product in the cart.